components/HeaderWithMenu/ConnectedWallets.tsx

```diff
@@ -1,38 +1,40 @@
 import CoinbaseIcon from "../icons/Wallets/Coinbase"
 import MetaMaskIcon from "../icons/Wallets/MetaMask"
 import RainbowIcon from "../icons/Wallets/Rainbow"
 import WalletConnectIcon from "../icons/Wallets/WalletConnect"
 
 import WalletIcon from "../icons/WalletIcon"
+import shortenAddress from "../utils/ShortenAddress"
 import BitKeep from "../icons/Wallets/BitKeep"
 import Argent from "../icons/Wallets/Argent"
 
 
-export const RainbowKitConnectWallet = ({ isButton }: { isButton?: boolean }) => {
+export const RainbowKitConnectWallet = ({ isButton, isMobile, isConnected, isMenuCard }: { isButton?: boolean, isMobile?: boolean, isConnected?: boolean, isMenuCard?: boolean }) => {
     return <ConnectButton.Custom>
         {({ openConnectModal, account, mounted, chain, openAccountModal }) => {
             const connected = !!(mounted && account && chain)
             const { connector } = useAccount()
-            
-            return <IconButton onClick={() => connected ? openAccountModal() : openConnectModal()} icon={
-                connected ?
-                    <div className="font-bold grow flex space-x-2">
-                        <div className="inline-flex items-center relative">
-                            <AddressIcon address={account.address} size={25} />
-                            {
-                                connector && <span className="absolute -bottom-1 -right-2 ml-1 shadow-sm text-[10px] leading-4 font-semibold text-white">
-                                    <ResolveWalletIcon connector={connector?.name} className="w-5 h-5 border-2 border-secondary-600 rounded-full bg-primary-text" />
-                                </span>
-                            }
+            return <button onClick={() => connected ? openAccountModal() : openConnectModal()} type="button" className={`${!isMobile && isMenuCard ? "h-24 w-24" : isMobile && isMenuCard ? "h-16 w-14" : ""} ${isMenuCard ? "mx-2 w-4/12 flex flex-col items-center justify-center border-2 border-secondary-500 menu-link rounded-md outline-none bg-secondary-700 text-primary-text hover:text-white" : "-mx-2 p-1.5 justify-self-start text-primary-text hover:bg-secondary-500 hover:text-white focus:outline-none inline-flex rounded-lg items-center"}`}>
+                {connected ?
+                    <div className="mx-0.5">
+                        <div className={`${isMenuCard ? "flex-col items-center" : "font-bold grow flex space-x-2"}`}>
+                            <div className="inline-flex items-center relative">
+                                <AddressIcon address={account.address} size={25} />
+                                {
+                                    connector && <span className="absolute -bottom-1 -right-2 ml-1 shadow-sm text-[10px] leading-4 font-semibold text-white">
+                                        <ResolveWalletIcon connector={connector?.name} className="w-5 h-5 border-2 border-secondary-600 rounded-full bg-primary-text" />
```

Comment:
I think we should have 2 or 3(if there's a lot of shared logic/UI) components for this instead of jamming everything in one.